libs/agno/agno/agent/agent.py

```diff
@@ -698,15 +699,22 @@ def _run(
                         else:
                             self.run_response.tools = tool_calls_list
 
-
                         reasoning_step: ReasoningStep = None
                         # For Reasoning/Thinking/Knowledge Tools update reasoning_content in RunResponse
                         if self.run_response.tools:
+                            reasoning_tool_calls = []
                             for tool_call in self.run_response.tools:
                                 tool_name = tool_call.get("tool_name", "")
                                 if tool_name.lower() in ["think", "analyze"]:
-                                    tool_args = tool_call.get("tool_args", {})
-                                    reasoning_step = self.update_reasoning_content_from_tool_call(tool_name, tool_args)
+                                    reasoning_tool_calls.append(tool_call)
+
+                            if len(reasoning_tool_calls) > 0:
+                                tool_args = reasoning_tool_calls[-1].get("tool_args", {})
```

Comment:
Hmmm... reasoning tools can be called multiple times at once though? This would always only take the "last" of such calls into account right? They could have different information?